sympy/functions/combinatorial/factorials.py

```diff
@@ -931,14 +931,14 @@ def _eval_Mod(self, q):
 
             # handle negative integers k or n
             if k < 0:
-                return 0
+                return S.Zero
             if n < 0:
                 n = -n + k - 1
                 res = -1 if k%2 else 1
 
             # non negative integers k and n
             if k > n:
-                return 0
+                return S.Zero
```

Comment:
I'd expect some test cases to pinpoint to execute these lines as well.
See the coverage report

Reply:
Thanks for reviewing this PR. I think it's good to go now.